src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -96,14 +99,26 @@ public void configure() {
                 .to("direct:insights");
 
         from("direct:insights").id("call-insights-upload-service")
-                .process(this::createMultipartToSendToInsights)
-                .setHeader(Exchange.HTTP_METHOD, constant(org.apache.camel.component.http4.HttpMethods.POST))
-                .setHeader("x-rh-identity", method(MainRouteBuilder.class, "getRHIdentity(${header.x-rh-identity}, ${header.CamelFileName}, ${headers})"))
-                .removeHeaders("Camel*")
-                .to("http4://" + uploadHost + "/api/ingress/v1/upload")
-                .to("log:INFO?showBody=true&showHeaders=true")
+                .doTry()
+                    .process(this::createMultipartToSendToInsights)
+                    .setHeader(Exchange.HTTP_METHOD, constant(org.apache.camel.component.http4.HttpMethods.POST))
+                    .setHeader("x-rh-identity", method(MainRouteBuilder.class, "getRHIdentity(${header.x-rh-identity}, ${header.CamelFileName}, ${headers})"))
+                    .removeHeaders("Camel*")
+                    .to("http4://" + uploadHost + "/api/ingress/v1/upload")
+                    .choice()
+                        .when(not(isResponseSuccess()))
+                            .throwException(org.apache.commons.httpclient.HttpException.class, "Unsuccessful response from Insights Upload Service")
+                .endDoTry()
+                .doCatch(Exception.class)
+                    .to("log:error?showCaughtException=true&showStackTrace=true")
+                    .to("direct:mark-analysis-fail")
                 .end();
 
+        from("direct:mark-analysis-fail").id("markAnalysisFail")
+                .process(e -> analysisService.updateStatus(AnalysisService.STATUS.FAILED.toString(), Long.parseLong((String) e.getIn().getHeader("MA_metadata", Map.class).get(ANALYSIS_ID))))
```

Comment:
I'm wondering if such a general purpose method could be add as `DefaultErrorHandler` as well so that, for whatever other use case not covered in this PR, we know we have it managed implementing something based on `DefaultErrorHandler`'s [example](https://camel.apache.org/manual/latest/defaulterrorhandler.html#DefaultErrorHandler-Example) like (consider it like pseudo-code):
```java
onException(Exception.class)
        .handled(true)
        .process(e -> analysisService.updateStatus(AnalysisService.STATUS.FAILED.toString(), Long.parseLong((String) e.getIn().getHeader("MA_metadata", Map.class).get(ANALYSIS_ID))));
```

Reply:
the problem is that we could have an exception when we still dont have analysisid in the route ( uploading the file to xavier-integration, or receiving the message from kafka, etc..... )

Reply:
i see your point and when we haven't got the analysis `id` it would fail, thanks